### PR TITLE
add file logic

### PR DESCRIPTION
**Pull request by mfdeveloper508**
general IRS file logic

### CODE REVIEW

File: pkg/file/file_instance.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	bufSize := len(buf)
+	readPtr := 0
+	if string(buf[readPtr]) != config.TRecordType || bufSize < config.RecordLength {
+		return utils.ErrInvalidAscii
+	}
+
+	if f.Transmitter == nil {
+		f.Transmitter = records.NewTRecord()
+	}
+	err := f.Transmitter.Parse(buf[readPtr : readPtr+config.RecordLength])
```
Comment (adamdecaf) [inline pkg/file/file_instance.go:44]: Can we add a length check on this?

File: pkg/file/file_instance.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	if string(buf[readPtr]) != config.FRecordType || bufSize < readPtr+config.RecordLength {
+		return utils.ErrInvalidAscii
+	}
+
+	if f.EndTransmitter == nil {
+		f.EndTransmitter = records.NewFRecord()
+	}
+	err = f.EndTransmitter.Parse(buf[readPtr : readPtr+config.RecordLength])
```
Comment (adamdecaf) [inline pkg/file/file_instance.go:68]: Length check.

File: pkg/file/person.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+	readPtr := 0
+
+	if string(buf[readPtr]) != config.ARecordType || bufSize < config.RecordLength {
+		return readPtr, utils.ErrInvalidAscii
+	}
+
+	if p.Payer == nil {
+		p.Payer = records.NewARecord()
+	}
+	err := p.Payer.Parse(buf[readPtr : readPtr+config.RecordLength])
```
Comment (adamdecaf) [inline pkg/file/person.go:103]: Length check.

File: pkg/file/person.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	p.Payees = []records.Record{}
+	for string(buf[readPtr]) == config.BRecordType {
+		if bufSize < readPtr+config.RecordLength {
+			return readPtr, utils.ErrInvalidAscii
+		}
+
+		newPayee := records.NewBRecord(typeOfReturn)
+		if err = newPayee.Parse(buf[readPtr : readPtr+config.RecordLength]); err != nil {
```
Comment (adamdecaf) [inline pkg/file/person.go:121]: Length check

File: pkg/file/person.go [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+			return readPtr, err
+		}
+
+		readPtr += config.RecordLength
+		p.Payees = append(p.Payees, newPayee)
+	}
+
+	if string(buf[readPtr]) == config.CRecordType || bufSize < readPtr+config.RecordLength {
```
Comment (adamdecaf) [inline pkg/file/person.go:129]: Length check

File: pkg/file/person.go [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	p.States = []records.Record{}
+	for string(buf[readPtr]) == config.KRecordType {
+		if bufSize < readPtr+config.RecordLength {
+			return readPtr, utils.ErrInvalidAscii
+		}
+
+		newState := records.NewKRecord()
+		if err = newState.Parse(buf[readPtr : readPtr+config.RecordLength]); err != nil {
```
Comment (adamdecaf) [inline pkg/file/person.go:147]: Length check

File: pkg/file/person.go [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+		p.Payer = records.NewARecord()
+		err := readJsonWithRecord(p.Payer, record)
+		if err != nil {
+			return err
+		}
+	}
+
+	typeOfReturn := ""
+	if p.Payer != nil {
+		typeOfReturn = config.TypeOfReturns[p.Payer.(*records.ARecord).TypeOfReturn]
```
Comment (adamdecaf) [inline pkg/file/person.go:180]: Can this return records of a type besides `ARecord`?

Comment (mfdeveloper508) [reply pkg/file/person.go:180]: no, we should use ARecord from p.Payer only
there isn't type convert check for p.Payer.(*records.ARecord). it seems that the check is unmeaningful

Comment (adamdecaf) [reply pkg/file/person.go:180]: That's true. It worries me to have unchecked casts even if the `ARecord.Parse(..)` would only return that type. It calls out to `utils.ParseValue` which might get accidentally broken in the future. 

I added a type check on the cast in https://github.com/moov-io/irs/commit/85db345624afb0b66aaba7844ddcbb499ba69d8c

File: pkg/file/file_instance.go (hunk 1 above)
Comment (mfdeveloper508) [reply pkg/file/file_instance.go:44]: there is length check

Comment (adamdecaf) [reply pkg/file/file_instance.go:44]: I don't see a check for `readPtr+config.RecordLength < len(buf)`

**Pull request merged**